### PR TITLE
ui - display all domains that reference current user

### PR DESCRIPTION
**Pull request by ArtjomsPorss**
# Description
Change to UI domains list for current user to include domains with indirect membership, e.g. domains that assume delegated roles.

# Contribution Checklist:
- [x] **The pull request does not introduce any breaking changes**
- [x] **I have read the [contribution guidelines](https://github.com/AthenZ/athenz/blob/master/CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request.**

### CODE REVIEW

File: ui/src/__tests__/redux/reducers/domains.test.js [hunk 1]
```diff
@@ -56,18 +56,28 @@ const userDomains = [
...
         };
         const expectedState = {
             domainsList: userDomains,
         };
         const newState = domains(initialState, action);
-        expect(_.isEqual(newState, expectedState)).toBeTruthy();
+        console.log(newState);
+        console.log(expectedState);
+        expect(_.isEqual(_.sortBy(newState), _.sortBy(expectedState))).toBeTruthy();
     });
```
Comment (ArtjomsPorss) [inline ui/src/__tests__/redux/reducers/domains.test.js:81]: fixed reducer test

File: ui/src/api.js [hunk 2]
```diff
@@ -36,11 +36,11 @@ const Api = (req) => {
...
 
-        listUserDomains(roleName) {
+        listUserDomains() {
             return new Promise((resolve, reject) => {
                 fetchr
-                    .read('domain-list')
-                    .params({ roleName })
+                    .read('domain-role-member')
+                    .params({  expand: true })
                     .end((err, data) => {
```
Comment (ArtjomsPorss) [inline ui/src/api.js:44]: using different api to obtain domain details. not passing principal as it is picked up from the request data

File: ui/src/redux/reducers/domains.js [hunk 3]
```diff
@@ -272,3 +272,46 @@ export const domains = (state = {}, action) => {
...
+            name: domain,
+            adminDomain: false
+        })
+    })
+
+    // sorting domains alphabetically
+    domainsToReturn.sort((a, b) => a.name.localeCompare(b.name));
+
+    return domainsToReturn;
+};
```
Comment (ArtjomsPorss) [inline ui/src/redux/reducers/domains.js:317]: api response transformation logic

File: ui/src/__tests__/redux/reducers/domains.test.js [hunk 4]
```diff
@@ -56,18 +56,26 @@ const userDomains = [
...
+                domainsList: payloadDomainsData,
             },
         };
         const expectedState = {
             domainsList: userDomains,
         };
         const newState = domains(initialState, action);
-        expect(_.isEqual(newState, expectedState)).toBeTruthy();
+        expect(_.isEqual(_.sortBy(newState), _.sortBy(expectedState))).toBeTruthy();
     });
```
Comment (ArtjomsPorss) [inline ui/src/__tests__/redux/reducers/domains.test.js:79]: fixed reducer test

File: ui/src/redux/thunks/domains.js [hunk 5]
```diff
@@ -266,3 +267,48 @@ export const processPendingMembers =
...
+            name: domain,
+            adminDomain: false
+        })
+    })
+
+    // sorting domains alphabetically
+    domainsToReturn.sort((a, b) => a.name.localeCompare(b.name));
+
+    return domainsToReturn;
+};
```
Comment (ArtjomsPorss) [inline ui/src/redux/thunks/domains.js:313]: according to docs thunks is the preferred location for data transformation, not the reducers

**Comment by ArtjomsPorss** [comment]
By changing API call - the data format was changed that is returned thus breaking the snapshot generation.
Multiple tests had to be fixed because they seem to be used by jest as a base for generating snapshots.

**Comment by ArtjomsPorss** [comment]
js file reformatting was done on some files unrelated to pr because these files were failing github build

File: ui/src/__tests__/components/utils/ReviewUtils.test.js
Comment (ArtjomsPorss) [inline ui/src/__tests__/components/utils/ReviewUtils.test.js:1]: only formatting change

File: ui/src/__tests__/spec/tests/domain.spec.js
Comment (ArtjomsPorss) [inline ui/src/__tests__/spec/tests/domain.spec.js:1]: only formatting change

File: ui/src/__tests__/spec/tests/groups.spec.js
Comment (ArtjomsPorss) [inline ui/src/__tests__/spec/tests/groups.spec.js:1]: only formatting change

**Review by chandrasekhar1996 — APPROVED** [review]

**Pull request merged**